configargparse.py

```diff
@@ -241,6 +241,18 @@ def parse_known_args(self, args = None, namespace = None,
         else:
             args = list(args)
 
+        new_args = list()
+        long_arg_re = re.compile(r"^(?P<key>--[^=]+)=(?P<value>.+)$")
```

Comment:
This regexp fails on short args like `-k=3`, and also if prefix chars get set to something other than `-`. It would be good to instead use `key, value =  arg.split('=', 1)` like in 
https://code.google.com/p/argparse/source/browse/argparse.py#2081